server/src/internalClusterTest/java/org/elasticsearch/snapshots/SnapshotStressTestsIT.java

```diff
@@ -803,12 +803,28 @@ private void startCleaner() {
                         .cluster()
                         .prepareCleanupRepository(trackedRepository.repositoryName)
                         .execute(mustSucceed(cleanupRepositoryResponse -> {
-                            Releasables.close(releaseAll);
-                            logger.info("--> completed cleanup of [{}]", trackedRepository.repositoryName);
                             final RepositoryCleanupResult result = cleanupRepositoryResponse.result();
-                            assertThat(Strings.toString(result), result.blobs(), equalTo(0L));
-                            assertThat(Strings.toString(result), result.bytes(), equalTo(0L));
-                            startCleaner();
+                            if (result.bytes() > 0L || result.blobs() > 0L) {
```

Comment:
Making this more robust might be interesting but I wonder if it's worth it in the real world. Technically, it wouldn't be so hard to make a data node fire off a bunch of deletes for the stuff it just wrote in case it runs into an abort I guess. I'm not sure the added complexity is worth it given how little data this tends to leak in the real world in practice, but it sure would make aborts cleaner in theory. I think I like the idea :) should I do that in a separate PR maybe?

> If not could we at least do a delete before the no-op cleanup (after acquiring all the permits) instead of the first cleanup?

It's tricky right? We'd have to do a delete that touches all the indices that could have something leaked in them. Which I guess (unless we're doing something really tricky) means we'd have to run a delete that touches a set of snapshots such that all indices in the repo are touched? It shouldn't be endlessly hard to do that but is it really worth it, given that we have a bunch of focused tests around deletes and not leaking blobs elsewhere?